File: aplicação/backend/src/main/java/eaproject/output/BaseOutput.java
package eaproject.output;

import com.fasterxml.jackson.annotation.JsonProperty;
import eaproject.base.FeedbackMessage;
import eaproject.enums.FeedbackSeverity;

import java.util.ArrayList;

public abstract class BaseOutput {
    private  ArrayList<FeedbackMessage> feedbackMessages = new ArrayList<>();

    @JsonProperty
    public ArrayList<FeedbackMessage> getFeedbackMessages() {
        return feedbackMessages;
    }

    public void setFeedbackMessages(ArrayList<FeedbackMessage> feedbackMessages) {
        this.feedbackMessages = feedbackMessages;
    }

    public void addFeedbackMessage(String message, FeedbackSeverity severity) {
        this.feedbackMessages.add(new FeedbackMessage(message, severity));
    }
}

File: aplicação/frontend/ea-project/src/models/output/BaseOutput.js
import FeedbackMessage from '@/models/base/FeedbackMessage';

export default class BaseOutput {
  constructor(feedbackMessages = []) {
    this.feedbackMessages = [];
    this.setFeedbackMessages(feedbackMessages);
  }

  setFeedbackMessages(feedbackMessages) {
    this.feedbackMessages = feedbackMessages.map(
      (msg) => new FeedbackMessage(msg.message, msg.severity)
    );
  }

  addFeedbackMessage(message, severity) {
    this.feedbackMessages.push(new FeedbackMessage(message, severity));
  }
}
